### DISCUSSION TITLE
How to add a data atrribute with a boolean value on the body tag

### DISCUSSION THREAD

**Discussion by Kobee1203** [nuxt/nuxt#18514] [Questions]
I try to add a data attribute with a boolean value on the body tag but the data attribute is not added correctly.

I create a server plugin that calls `useHead` to add attributes on the body tag:

Here is an example:

```
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.hook('vue:setup', () => {
    useHead({
      title: 'This is my title',
      bodyAttrs: {
        'data-boolean': true,
        'data-boolean-str': "true",
        'data-boolean-str-uppercase': 'TRUE',
        'data-boolean-false': false,
        'data-boolean-false-str': "false",
        'data-boolean-false-str-uppercase': 'FALSE',
        'data-number': 1234,
        'data-yesno': 'yes',
        'data-1': 1,
        'data-1-str': '1',
      },
    });
  });
});
```

But the previous code gives the following result:

![image](https://user-images.githubusercontent.com/1423193/214573214-1bb8e9d1-edd4-4023-9842-82eb9c2073b5.png)

- `data-boolean` is added but with an empty value
- `data-boolean-str` is added but with an empty value
- `data-boolean-false` is not added
- `data-boolean-false-str` is not added
- Other data attributes are correctly added on the body tag

Do you have a solution to add a data attribute with a boolean value?

Here is a stackblitz project that shows the behavior:

https://stackblitz.com/edit/nuxt-starter-h5l6c8?file=plugins/head.server.ts

**Comment by ndragun92** [comment]
It definitely behaves as an error but as you can see using true/false as boolean/string is actually adding or removing data attribute which means that you can still check if data attribute is present since if its true it will be rendered in body tag if it is false it won't be rendered
